src/main/java/com/google/api/generator/gapic/composer/ServiceClientSampleCodeComposer.java

```diff
@@ -169,4 +179,169 @@ public static String composeClassHeaderEndpointSampleCode(
+        .map(
+            arg -> {
+              VariableExpr argVarExpr =
+                  VariableExpr.withVariable(
+                      Variable.builder()
+                          .setName(JavaStyle.toLowerCamelCase(arg.name()))
+                          .setType(arg.type())
+                          .build());
+              if (arg.type().equals(TypeNode.STRING)
+                  && arg.field().hasResourceReference()
+                  && resourceNames.containsKey(
+                      arg.field().resourceReference().resourceTypeString())) {
+                return MethodInvocationExpr.builder()
+                    .setExprReferenceExpr(argVarExpr)
+                    .setMethodName("toString")
+                    .build();
+              }
+              return argVarExpr;
+            })
+        .collect(Collectors.toList());
+  }
+
+  // Return a list of AssignmentExpr for method argument with its default value.
+  private static List<Expr> assignMethodArgumentsWithDefaultValues(
+      List<MethodArgument> arguments,
+      List<Expr> argVarExprs,
+      Map<String, ResourceName> resourceNames) {
+    List<ResourceName> resourceNameList =
+        resourceNames.values().stream().collect(Collectors.toList());
+    List<Expr> assignmentExprs = new ArrayList<>();
+    for (int i = 0; i < arguments.size(); i++) {
+      MethodArgument arg = arguments.get(i);
+      VariableExpr argVarExpr =
+          (argVarExprs.get(i) instanceof VariableExpr) ? (VariableExpr) argVarExprs.get(i) : null;
+      Expr defaultValueExpr = DefaultValueComposer.createDefaultValue(arg, resourceNames);
+      if (arg.type().equals(TypeNode.STRING)
+          && arg.field().hasResourceReference()
+          && resourceNames.containsKey(arg.field().resourceReference().resourceTypeString())) {
+        ResourceName resourceName =
+            resourceNames.get(arg.field().resourceReference().resourceTypeString());
+        defaultValueExpr =
+            DefaultValueComposer.createDefaultValue(
+                resourceName, resourceNameList, arg.field().name());
+        TypeNode resourceReferenceType =
+            arg.field().resourceReference().isChildType()
+                ? TypeNode.withReference(
+                    ConcreteReference.withClazz(com.google.api.resourcenames.ResourceName.class))
+                : defaultValueExpr.type();
+        argVarExpr =
+            VariableExpr.withVariable(
+                Variable.builder()
+                    .setName(JavaStyle.toLowerCamelCase(arg.name()))
+                    .setType(resourceReferenceType)
+                    .build());
+      }
+      assignmentExprs.add(
+          AssignmentExpr.builder()
+              .setVariableExpr(argVarExpr.toBuilder().setIsDecl(true).build())
+              .setValueExpr(defaultValueExpr)
+              .build());
+    }
+    return assignmentExprs;
+  }
+
+  private static Expr createAssignExprForVariableWithClientMethod(
```

Comment:
Can we inline this into the call site since there's only one usage? If we need to reuse it in a future PR, let's add this method only then, as that will help us pick a more generic name.